### PR TITLE
Adding Travis for testing, fixing minor PIP violations.

### PR DESCRIPTION
**Pull request by joshmarshall**
In order to support other people actually contributing sanely, I've added the first step of Travis integration so it runs the (minimal) existing tests in both 2.6 and 2.7. As I went through I made minor whitespace corrections and any test changes necessary to run with nosetests in both unittest and unittest2 test cases.

Feedback desired!

### CODE REVIEW

**Comment by efokschaner** [comment]
:+1: 
The changes look fine.

I had a few observations about the tests for the future:
1. You could make the test server port dynamic (in the ephemeral range) by binding to `0` and then retrieving it with `socket.getsockname()` which nicely avoids any potential for conflict.
2. I notice that a lot of the validation is done through the `history` system. It would be a more "black box" test to verify the client using the raw data that the server receives.
3. To reduce the likelihood of non-compliance in the server masking non-compliance in the client, you could avoid running an implementation of jsonrpc on the server and instead just use some pretty static response logic. This would then also allow you to easily have less "happy-path" tests where the server intentionally responds abnormally. The drawback with this route is you'd probably then want separate tests for your server implementation.

If you want, I can apply the above ideas in the tests that I'll be adding for the _Method changes unless you'd prefer consistency with the other tests. Let me know.

File: tests.py [hunk 1]
```diff
@@ -1,16 +1,16 @@
 """
 The tests in this file compare the request and response objects
 to the JSON-RPC 2.0 specification document, as well as testing
-several internal components of the jsonrpclib library. Run this 
+several internal components of the jsonrpclib library. Run this
 module without any parameters to run the tests.
 
-Currently, this is not easily tested with a framework like 
+Currently, this is not easily tested with a framework like
```
Comment (efokschaner) [inline tests.py]: This paragraph is a little contradictory now that you are using nose.

**Comment by joshmarshall** [comment]
@efokschaner 

You bring up several very good points -- to be honest, this project has remained mostly unchanged for years, and I wrote it pretty early in my Python learning. :) It needs a fairly thorough overhaul to be brought up to a decent standard. Your notes highlight some of that. I've made a couple of the changes (fixed the message, and moved to dynamic ports) you recommended, if you'd like to make some tweaks yourself I certainly would appreciate it!

That being said, your change is currently blocked by this PR, and this PR would help move the project into a healthier process (if not fixing some of the issues themselves). Whenever we feel like this is "good enough" to move to next step, we should probably do that and tackle next pieces.

Thanks!

**Comment by efokschaner** [comment]
@joshmarshall I fully agree this is good enough. My notes were just ideas for the future. I added a comment in the source regarding the port stuff.

I'm still :+1: on this PR with or without any further changes.

**Pull request merged**